crates/oxc_ast/src/ast/js.rs

```diff
@@ -2562,3 +2565,14 @@ pub enum ModuleExportName<'a> {
     IdentifierReference(IdentifierReference<'a>) = 1,
     StringLiteral(StringLiteral<'a>) = 2,
 }
+
+/// Intrinsics in Google's V8 engine `%GetOptimizationStatus`.
+/// See: https://github.com/v8/v8/blob/5fe0aa3bc79c0a9d3ad546b79211f07105f09585/src/runtime/runtime.h#L43
+#[ast(visit)]
+#[derive(Debug)]
+#[generate_derive(CloneIn, GetSpan, GetSpanMut, ContentEq, ESTree)]
+pub struct V8IntrinsicExpression<'a> {
+    pub span: Span,
+    pub name: IdentifierReference<'a>,
```

Comment:
I think this should be `IdentifierName` not `IdentifierReference`.

```js
const Foo = 123;
%Foo(123);
```

`Foo` in `%Foo` does not act as a reference to `const Foo`.

V8 intrinsic names should probably be considered a separate namespace from symbols, in the same way as:

* `label` in `label: while (whatever) { break label; }`
* `div` in JSX `<div>stuff</div>`

Reply:
Yep. Already fixed in the followup.
This is in a WIP state, I can drop a ping in the discord server when it's ready, if that helps :)

Reply:
Note: `Foo` in `<Foo>` is an `IdentifierReference`, but `div` in `<div>` is an `IdentifierName`, because it doesn't refer to variable called `div`.

Reply:
Sorry, not trying to step on your toes before you're ready. Just I was interested, so I took a look.